src/Symfony/Component/DependencyInjection/Tests/Compiler/PriorityTaggedServiceTraitTest.php

```diff
@@ -199,3 +230,8 @@ public function test($tagName, ContainerBuilder $container)
         return $this->findAndSortTaggedServices($tagName, $container);
     }
 }
+
+#[TaggedItem(index: 'hello', priority: 1)]
```

Comment:
sidenote, we should document the power of `index: self::class` :)

Reply:
the default index in the service id, which usually is also the FQCN!

Reply:
but how does `public string $index` take a default?

Reply:
It's the default managed by `PriorityTaggedServiceTrait`

Reply:
i meant, end users can save themselves from boring "naming things" by leveraging `index: self::class` :)

Reply:
but to do that, you would just omit the index entirely in the attribute, letting it be `null` and so triggering the default in PriorityTaggedServiceTrait

Reply:
Right, see #40484